go-controller/pkg/ovn/default_network_controller.go

```diff
@@ -727,6 +727,26 @@ func (h *defaultNetworkControllerEventHandler) AddResource(obj interface{}, from
 		}
 		return h.oc.ensurePod(nil, pod, true)
 
+	case factory.PolicyType:
+		np, ok := obj.(*knet.NetworkPolicy)
+		if !ok {
+			return fmt.Errorf("could not cast %T object to *knet.NetworkPolicy", obj)
+		}
+
+		netinfo, err := h.oc.getActiveNetworkForNamespace(np.Namespace)
+		if err != nil {
+			return fmt.Errorf("could not get active network for namespace %s: %v", np.Namespace, err)
+		}
+		if !netinfo.IsDefault() {
+			return nil
+		}
```

Comment:
Why don't you do this in `base_network_controller.go`?

You would just have to check `bnc.GetNetwokrName() != netinfo.GetNetworkName()`

How would you otherwise prevent the netpol that is processed by the default network controller to be processed as well on primary network controllers?

Reply:
ok @jcaamano , having this in  `base_network_controller.go`  check is a good idea. done.
one question though: secondary network controllers never handle default network event ? correct ?

Reply:
No sure what you mean. Non primary but still secondary networks don't handle events for `factory.PolicyType` because the retry framework should not be setup for them.